lib/shoppertrak_api_client.py

```diff
@@ -0,0 +1,161 @@
+        rows = []
+        for site_xml in xml_root.findall("site"):
+            site_str = self._get_xml_str(site_xml, "siteID")
+            for date_xml in site_xml.findall("date"):
+                date_str = self._get_xml_str(date_xml, "dateValue")
+                if date_str != input_date_str:
+                    self.logger.error(
+                        f"Request date does not match response date.\nRequest date: "
+                        f"{input_date_str}\nResponse date: {date_str}"
+                    )
+                    raise ShopperTrakApiClientError(
+                        f"Request date does not match response date.\nRequest date: "
+                        f"{input_date_str}\nResponse date: {date_str}"
+                    )
+                for entrance_xml in date_xml.findall("entrance"):
+                    entrance_val = self._get_xml_str(entrance_xml, "name")
+                    if entrance_val:
+                        entrance_val = self._cast_str_to_int(entrance_val.lstrip("EP"))
+                    for traffic_xml in entrance_xml.findall("traffic"):
+                        enters = self._cast_str_to_int(
+                            self._get_xml_str(traffic_xml, "enters")
+                        )
+                        exits = self._cast_str_to_int(
+                            self._get_xml_str(traffic_xml, "exits")
+                        )
+
+                        start_time_str = self._get_xml_str(traffic_xml, "startTime")
+                        start_dt_str = datetime.strptime(
+                            date_str + " " + start_time_str, "%Y%m%d %H%M%S"
+                        ).isoformat()
+
+                        code = self._get_xml_str(traffic_xml, "code")
+                        is_healthy_orbit = code == "01"
+                        if code != "01" and code != "02":
+                            self.logger.warning(
+                                f"Unknown code: '{code}'. Setting is_real to False"
+                            )
+
+                        if is_healthy_orbit or not is_recovery_mode:
+                            rows.append(
+                                {
+                                    "shoppertrak_site_id": site_str,
+                                    "orbit": entrance_val,
+                                    "increment_start": start_dt_str,
+                                    "enters": enters,
+                                    "exits": exits,
+                                    "is_healthy_orbit": is_healthy_orbit,
+                                    "is_recovery_data": is_recovery_mode,
+                                    "poll_date": self.today_str,
+                                }
+                            )
+        return rows
+
+    def _check_response(self, response_text):
+        try:
+            root = ET.fromstring(response_text)
+            error = root.find("error")
+        except Exception as e:
```

Comment:
Any way to make this error handling more specific?

Reply:
Changed to ElementTree.ParseError